src/components/ProductDetail/ProductDetailOrderForm.js

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useState } from "react";
+import SizeOptionsContainer from "./SizeOptionsContainer";
+import { camelToPascalCase } from "../../helperFunctions";
+import { Link } from "react-router-dom";
+
+const ProductDetailOrderForm = ({product, addToShoppingBag, shoppingBag, updateQuantity}) => {
+  const [isSingleSize, setIsSingleSize] = useState(false);
+  const [inputFields, setInputFields] = useState({
+    "color":"",
+    "size": "",
+    "quantity":0
+  });
+  const [colorOptions, setColorOptions] = useState([]);
+  const [isFormHealthy, setIsFormHealthy] = useState(false);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+  
+  useEffect(() => {
+    setColorOptions(product?.colorOptions)
+    console.log(product)
+    if (product.sizeOptions?.length === 1) {
+      setIsSingleSize(true);
+    } else {
+      setIsSingleSize(false);
+    }
+  }, [product]);
+  
+  
+  useEffect(() => {
+    checkFormHealth()
+  }, [inputFields]);
+  
+  useEffect(() => {
+    const total = shoppingBag.reduce((total, item) => total + item.quantity, 0);
+    setTotalQuantity(total);
+  }, [shoppingBag])
+  
+  
+  const checkFormHealth = () => {
+    if (inputFields.color && inputFields.size && inputFields.quantity) {
+      setIsFormHealthy(true);
+    } else {
+      setIsFormHealthy(false);
+    }
+  }
+
+  const handleSelect = (e, changedField) => {
+    const clonedInputs = {...inputFields};
+    clonedInputs[changedField] = e.target.value;
+    setInputFields(clonedInputs);
+    checkFormHealth();
+  }
+
+  const ColorOptionsCode = () => {
+    const options = colorOptions?.map(color => {
+      return <option key={color} value={color}> { camelToPascalCase(color) } </option>
+    })
+    return (
+      <>
+        <option value="">Choose color</option>
+        {options}
+      </>
+    );
+  }
+  
+  const checkForExistingMatch = (cart, item) => {
```

Comment:
So this checks for existing items to add to quantity? Very cool.

Reply:
Correct. If there's a match, it does not add a new product to the cart. Only adds to the quantity of that existing product.